main.cpp

```diff
@@ -258,271 +282,316 @@ struct MotorCycle
+
+    return percentGasLevel;
+}
+
+bool MotorCycle::cruise(float speed)
+{
+    isCruising = true;
+    revolutionsPerMinute = speed;
+
+    return isCruising;
+}
+
+void MotorCycle::getNewWheels(MotorCycle::Wheel wheelA)
+{
+    frontWheel = wheelA;
+    backWheel = wheelA;
+}
+
 struct DrillSet
 {
-    // drill bit material
     std::string drillbitMaterial = "Steel";
-    // drill bit diameter
     unsigned int drillbitDiameter = 2;
-    // angular speed
     float angularSpeed = 0.0f;
-    // number of chargers
     unsigned int numberOfChargers = 2;
-    // brand name
     std::string brandName = "Bosch";
+    int directionOfRotation = 1;
 
-    // can the drill bit drill concrete?
     bool canDrillConcrete(std::string drillbitMaterial); // confirm whether or not the drill bit can drill concrete
-    // unscrew in the left or right direction
     void setDirectionOfRotation(int direction = -1); // left= -1, right = 1
-    // set rotatory speed
     void setRotatorySpeed(float angularSpeed = 2.0f);
     
 };
 
-/*
-Thing 5) Speakers
-5 properties:
-    1) loudness (float)
-    2) resistance (unsigned int)
-    3) distortion (float)
-    4) diameter (double)
-    5) amount of power required in mW (unsigned int)
-3 things it can do:
-    1) play sound
-    2) consume power
-    3) set playback level
- */
+bool DrillSet::canDrillConcrete(std::string drillbitMaterialA)
+{
+    if(drillbitMaterialA == "Steel")
+        return true;
+
+    return false;
+}
+
+void DrillSet::setDirectionOfRotation(int direction)
+{
+    directionOfRotation = direction;
+}
+
+void DrillSet::setRotatorySpeed(float angularSpeedA)
+{
+    angularSpeed = angularSpeedA;
+}
 
 struct Speaker
 {
-    // loudness
     float loudness = 90.0f;
-    // resistance
-    unsigned int resistance = 4;
-    // distortion
+    float resistance = 4.0f;
     float distortion = 0.1f;
-    // diameter
     double diameter = 20.1;
-    // amount of power required in mW
     unsigned int powerAmount = 10;
+    bool isPlaying = false;
 
-    // play sound
     void playSound();
-    // power consumed
-    float powerConsumption(unsigned int resistance = 4, float voltageInput = 1.0f);
-    // set playback level
+    float powerConsumption(float voltageInput = 1.0f);
     void setPlaybackLevel(unsigned int volumeIndex);
 
 };
 
-/*
-Thing 6) Microphones
-5 properties:
-    1) sensitivity (float)
-    2) power supply (unsigned int)
-    3) acoustic overload point (float)
-    4) technology (string)
-    5) transduction type (string)
-3 things it can do:
-    1) capture sound
-    2) set input gain
-    3) read sensitivity
- */
+void Speaker::playSound()
+{
+    isPlaying = true;
+}
+
+float Speaker::powerConsumption(float voltageInput)
+{
+    float power;
+
+    power = ( voltageInput * voltageInput ) / resistance;
+
+    return power;
+}
+
+void Speaker::setPlaybackLevel(unsigned int volumeIndex)
+{
+    loudness *= static_cast<float>(volumeIndex);
+}
```

Comment:
Casting has not been covered yet. Remove this cast and change your code to remove any warnings about implicit casting.
- [ ] revise